receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -48,23 +55,72 @@ type statsDMetric struct {
 	labelValues      []*metricspb.LabelValue
 }
 
+func (p *StatsDParser) Initialize() error {
+	gauges = make(map[string]*metricspb.Metric)
+	counters = make(map[string]*metricspb.Metric)
+	return nil
+}
+
+// get the metrics preparing for flushing and reset the state
+func (p *StatsDParser) GetMetrics() []*metricspb.Metric {
+	var metrics []*metricspb.Metric
+
+	for _, metric := range gauges {
+		metrics = append(metrics, metric)
+	}
+
+	for _, metric := range counters {
+		metrics = append(metrics, metric)
+	}
+
+	gauges = make(map[string]*metricspb.Metric)
+	counters = make(map[string]*metricspb.Metric)
+
+	return metrics
+}
+
 var timeNowFunc = func() int64 {
 	return time.Now().Unix()
 }
 
-// Parse returns an OTLP metric representation of the input StatsD string.
-func (p *StatsDParser) Parse(line string) (*metricspb.Metric, error) {
+//aggregate for each metric line
+func (p *StatsDParser) Aggregate(line string) error {
 	parsedMetric, err := parseMessageToMetric(line)
```

Comment:
FYI there's a memory allocation here that could be avoided by not returning a pointer from `parseMessageToMetric`. I expect this will be a win because in the common case we expect to find a map entry and the `parsedMetric` value will be discarded. Eventually this sort of concern will matter in this code.

Reply:
Oh, I see. Thanks, will let `parseMessageToMetric` return a struct but not the pointer